src/pool/mempool.rs

```diff
@@ -140,58 +136,45 @@ impl<SP: starknet::providers::Provider + Send + Sync + Clone + 'static> AccountM
     }
 
     /// Returns the next available account from the manager.
-    pub async fn lock_account(&self) -> eyre::Result<LockedRelayer<'_, JsonRpcClient<HttpTransport>>>
+    pub async fn get_relayer(&self) -> eyre::Result<Relayer<JsonRpcClient<HttpTransport>>>
     where
         SP: starknet::providers::Provider + Send + Sync + Clone + 'static,
     {
         // Use `StdRng` instead of `ThreadRng` as it is `Send`
         let mut rng = rand::rngs::StdRng::from_entropy();
 
-        // Collect the accounts into a vector for shuffling
-        let mut accounts: Vec<_> = self.accounts.iter().collect();
+        // Shuffle indices of accounts randomly
+        let mut account_indices: Vec<_> = (0..self.accounts.len()).collect();
+        account_indices.shuffle(&mut rng);
 
-        // Shuffle the accounts randomly before iterating
-        accounts.shuffle(&mut rng);
+        for index in account_indices {
+            let account_address = self.accounts[index];
 
-        loop {
-            if accounts.is_empty() {
-                return Err(eyre::eyre!("failed to fetch funded account"));
-            }
+            // Retrieve the balance of the selected account
+            let balance = self.get_balance(account_address).await?;
 
-            // Create the future locks with indices for more efficient removal
-            // use [`select_all`] to poll an iterator over impl Future<Output = (Felt, MutexGuard<Felt>)>
-            // We use Box::pin because this Future doesn't implement `Unpin`.
-            let fut_locks = accounts
-                .iter()
-                .enumerate()
-                .map(|(index, (address, nonce))| Box::pin(async move { (index, *address, nonce.lock().await) }));
-
-            // Select the first account that gets unlocked
-            let ((index, account_address, guard), _, _) = select_all(fut_locks).await;
-
-            // Fetch the balance of the selected account
-            let balance = self.get_balance(*account_address).await?;
-
-            // If the balance is lower than the threshold, remove the account using swap_remove
+            // Skip accounts with insufficient balance
             if balance < U256::from(ONE_TENTH_ETH) {
-                accounts.swap_remove(index);
                 continue;
             }
 
-            let balance = into_via_try_wrapper!(balance)?;
+            // Retrieve chain ID for the Ethereum context
             let chain_id = self.eth_client.starknet_provider().chain_id().await?;
+            let balance = into_via_try_wrapper!(balance)?;
 
-            let account = LockedRelayer::new(
-                guard,
-                *account_address,
+            // Construct the `Relayer` with the account address and other relevant data
+            let account = Relayer::new(
+                account_address,
                 balance,
                 JsonRpcClient::new(HttpTransport::new(KAKAROT_RPC_CONFIG.network_url.clone())),
                 chain_id,
```

Comment:
let's remove the chain id from here, and in the `Relayer` itself, use `STARKNET_CHAIN_ID`

Reply:
yep, fixed